src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/Handlers/CommandLineItemHandler.cs

```diff
@@ -53,9 +53,11 @@ public override Task HandleAsync(IProjectVersionedValue<IProjectSubscriptionUpda
             if (!ProcessDesignTimeBuildFailure(projectChange, context))
             {
                 ProcessOptions(projectChange, context);
-                ProcessItems(projectChange, context, isActiveContext);
             }
 
+            // We need to process items even from failed design time builds, as we might be removing some items.
```

Comment:
I tried couple of different approaches here, but no avail:

1. Listen to changes to "TargetFramework" in LanguageServiceHost and force reset the WorkspaceProjectContext and CPS subscriptions.
2. Set "TargetFramework" as a project configuration dimension, even for projects targeting a single framework with explicit "TargetFramework" property (currently we do it only for cross targeting projects). This way, change to TargetFramework would change the project configuration, and hence require a new configured project to be loaded. However, after making this change, CPS doesn't seem to be generating evaluation updates when TargetFramework is edited, also the approach seems a bit hacky.